crates/core/src/test_utils/deploy_helpers.rs

```diff
@@ -469,17 +489,17 @@ async fn fund_eoa(
 /// Asynchronously deploys an Externally Owned Account (EOA) to the network and funds it.
 ///
 /// This function first computes the Starknet address of the EOA to be deployed using the provided
-/// account, contract address, and EOA account address. Then, it deploys the EOA to the network and
-/// funds it with the specified amount of fee token.
-async fn deploy_and_fund_eoa(
+/// account, contract address, and EOA account address. Then, it firstly funds the eoa to be able to
+/// pay for its deploymebt fee and then deploys the EOA to the network.
+async fn fund_and_deploy_eoa(
     account: &SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>,
     contract_address: FieldElement,
     amount: FieldElement,
     eoa_account_address: FieldElement,
     fee_token_address: FieldElement,
 ) -> FieldElement {
     let eoa_account_starknet_address = compute_starknet_address(account, contract_address, eoa_account_address).await;
-    fund_eoa(account, eoa_account_starknet_address, amount + *DEPLOY_FEE, fee_token_address).await;
+    fund_eoa(account, eoa_account_starknet_address, amount, fee_token_address).await;
```

Comment:
Why do we remove the deploy fee?

Reply:
- while using the function initially I got confused, because when I used it to fund and deploy an account, I was expecting it to have amount - DEPLOY_FEE as balance, so I thought anyone with the assumptions I have will get confused on this behaviour and would have to look inside this function
- so, I thought it is better if we remove it outside from this function, and if someone wants this, they can simply add *DEPLOY_FEE

Lemme know if you think otherwise.

Reply:
Right but imo people could also get confused it they set the amount to 0 and they are unable to deploy. Maybe something like 
```suggestion
    let amount = if amount < *DEPLOY_FEE { *DEPLOY_FEE } else { amount };
    fund_eoa(account, eoa_account_starknet_address, amount, fee_token_address).await;
```

Reply:
Missing update